news/_posts/2019-09-11-prestashop-in-2019-and-beyond-part-2-pain-points.md

```diff
@@ -0,0 +1,471 @@
+
+As platform developers, we face a big challenge that most developers (or at least, web developers) usually don't need to care about: **backwards compatibility**.
+
+Developers who work with PrestaShop (or any platform for that matter) expect things to work in a certain, predictable way. Either because the vendor explicitly described them that way through documentation, or because they spent time figuring out how it works on their own, in many cases through excruciating trial-and-error and reverse engineering of undocumented code. The bottom line is, once they get something working, developers expect it to _continue working that way_—and it's a natural thing to do.
+
+This expectation about how two independent pieces of software are supposed to work together is usually based on an _interface_, a kind of contract or agreement between the two parties, that can be either explicit (ie. clearly defined) or not. If it's implicit, then it's an _understanding_ of sorts, where one of the parties basically assumes the terms while the the other party does neither validate nor challenge them. 
+
+Of course, after having invested time and money in making something work, developers won't appreciate having to update their code regularly every time there's a new platform version—ideally, it should "just work". Therefore, it goes without saying that introducing any change that breaks this agreed-upon _interface_ (a "breaking change") won't help with keeping developers interested in adopting a new version of said platform. Again, this is true for _all_ platforms.
+
+As a result, many platform vendors—PrestaShop included—will go out of their way to avoid introducing breaking changes as much as possible. But this is a double-edged sword.
+
+It is well-known that, in time, [software rots](https://en.wikipedia.org/wiki/Software_rot). Most developers agree that the best medicine for avoiding software decay is applying frequent [refactoring](https://refactoring.com/), because once code rots for good, it becomes unmaintainable and has to be rewritten—and history shows us that [rewriting from scratch is a terrible idea](https://www.joelonsoftware.com/2000/04/06/things-you-should-never-do-part-i/).
+
+However, refactoring can only get you so far until it starts to require introducing breaking changes. And once you hit that brick wall, you're stuck: either you let that part of the code rot or you introduce breaking changes that will undermine the adoption of your platform. Or worse, you introduce an _new_ subsystem that does the same thing as the old one, and keep the previous one for compatibility, consequently increasing the overall software complexity.
+
+That's one of the top reasons PrestaShop's architecture has become so complex. The constant need to provide as much backwards compatibility as possible, even when doing so was detrimental to the platform's overall technical quality, has substantially undermined the project's capacity to move forward.
```

Comment:
This sentence is quite long, so it's a bit abstract. Maybe we could reaffirm the idea behind with a short behind ?

Like `PrestaShop code is complex because it did too much.`

Reply:
How about this?

```suggestion
That's one of the top reasons PrestaShop's architecture has become so complex. Because it was pushed too far. By constantly trying our best to maintain backwards compatibility, even when doing so was detrimental to the platform's overall technical quality, we have substantially undermined its capacity to move forward.
```

Reply:
👍